Package.swift

```diff
@@ -1,26 +1,59 @@
 // swift-tools-version:5.6
 
 import PackageDescription
+import Foundation
 
 var globalSwiftSettings: [PackageDescription.SwiftSetting] = []
+
+// Only enable if Swift 5.7+ is available and the environment variable `LOCALDEV` is
+// set to a value (such as 'true')
 #if swift(>=5.7)
-    globalSwiftSettings.append(.unsafeFlags(["-Xfrontend", "-strict-concurrency=complete"]))
-    /*
-     Summation from https://www.donnywals.com/enabling-concurrency-warnings-in-xcode-14/
-     Set `strict-concurrency` to `targeted` to enforce Sendable and actor-isolation
-     checks in your code. This explicitly verifies that `Sendable` constraints are
-     met when you mark one of your types as `Sendable`.
+    if ProcessInfo.processInfo.environment["LOCALDEV"] != nil {
+        /*
+        Summation from https://www.donnywals.com/enabling-concurrency-warnings-in-xcode-14/
+        Set `strict-concurrency` to `targeted` to enforce Sendable and actor-isolation
+        checks in your code. This explicitly verifies that `Sendable` constraints are
+        met when you mark one of your types as `Sendable`.
 
-     This mode is essentially a bit of a hybrid between the behavior that's intended
-     in Swift 6, and the default in Swift 5.7. Use this mode to have a bit of
-     checking on your code that uses Swift concurrency without too many warnings
-     and / or errors in your current codebase.
+        This mode is essentially a bit of a hybrid between the behavior that's intended
+        in Swift 6, and the default in Swift 5.7. Use this mode to have a bit of
+        checking on your code that uses Swift concurrency without too many warnings
+        and / or errors in your current codebase.
 
-     Set `strict-concurrency` to `complete` to get the full suite of concurrency
-     constraints, essentially as they will work in Swift 6.
-     */
+        Set `strict-concurrency` to `complete` to get the full suite of concurrency
+        constraints, essentially as they will work in Swift 6.
+        */
+        globalSwiftSettings.append(.unsafeFlags(["-Xfrontend", "-strict-concurrency=complete"]))
+    }
 #endif
 
+let FFIbinaryTarget: PackageDescription.Target
+// If either of the environment variables `CI` or `LOCALDEV` are set, then use
+// a local reference to an XCFramework file (built from `./scripts/build-xcframework.sh`)
+// instead of the previous released version.
+//
+// This script _does_ expect that you have Rust installed locally in order to function.
+if ProcessInfo.processInfo.environment["CI"] != nil || ProcessInfo.processInfo.environment["LOCALDEV"] != nil {
```

Comment:
What do you think about doing something like
```swift
let buildType = ProcessInfo.processInfo.environment["BUILD_TYPE"] ?? "LOCALDEV"
```
Where possible values of `BUILD_TYPE` are `LOCALDEV`, `CI` and `RELEASE`.
So we would essentially be able to write something like:
```swift
if buildType == "CI" || buildType == "LOCALDEV" {
   // with path
} 
if buildType == "RELEASE" {
  // with checksum
}
```
One if after another looks stupid, but that's why I like it – no negations, no nil checks – just dead simple condition with obvious execution paths.

Reply:
and I guess we will be able to just add 
```yml
env:
  BUILD_TYPE: CI
```
to `build.yml` file